src/tests/ScoreBoard.spec.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { render } from 'react-testing-library';
+import { ScoreBoard } from '../ScoreBoard';
+
+describe('ScoreBoard component', () => {
+  const mockUserMoves = [
+    { name: 'James', moves: 28 },
+    { name: 'Andy', moves: 30 },
+    { name: 'Bill', moves: 32 }
+  ];
+
+  it('Should render correctly', () => {
+    const { container } = render(<ScoreBoard userMoves={mockUserMoves} />);
+    expect(container.querySelectorAll('li')).toHaveLength(3)
```

Comment:
Also add assertions like these-

1. Check if actual text got rendered properly.
```js
expect(getNodeText(elem)).toBe('itaditya: 35 moves');
```

2. Test that if no data is present, nothing in the Scoreboard is rendered not even the Scoreboard heading.

Reply:
Sorry but do you have any example that I can follow?

Reply:
Take a look at this https://github.com/itaditya/trick-or-treat-game/blob/master/src/tests/Game.test.js#L11

Here we are returning in object in which there is a getItem function. The getItem function is actually doing nothing. Now in your test, what you have to do is make this function return a list of items

Something like this:

```js
const mockLocalStorage = {
  getItem: function (key) {
    if (key === 'userMoves') {
      return [{}, {}, {}]; // objects with data
    }

    return null;
  },
  setItem: jest.fn(),
  clear: jest.fn()
};
```

Also make sure you use `window.localStorage` instead of just localStorage everywhere just like I have done here https://github.com/itaditya/trick-or-treat-game/blob/master/src/Game.js#L55

Reply:
@itaditya Thanks, it working now, for anyone who wanted to try this, you need to JSON.stringify the array on getItem return

Reply:
Oh yeah, good catch @devTeaa!